src/features/staking/subcomponents/ApproveSuccess.tsx

```diff
@@ -1,79 +1,69 @@
-import { Dispatch, FC } from "react";
+import { FC } from "react";
 import { IoMdOpen } from "react-icons/io";
 import greenCheck from "../../../assets/check-green.svg";
-import { etherscanLink } from "../../../utils/constants";
-import { StatusStaking } from "../types/StakingTypes";
+import { StakeOrUnstake, WagmiLoadingStatus } from "../types/StakingTypes";
+import { etherscanLink } from "../utils/helpers";
 
 interface ApproveSuccessProps {
-  statusStaking: StatusStaking;
-  setStatusStaking: Dispatch<StatusStaking>;
-  amountApproved?: string;
-  amountStaked?: string;
+  stakeOrUnstake: "stake" | "unstake";
+  statusUnstake: WagmiLoadingStatus;
+  amount: string;
   chainId: number;
-  transactionHashApprove?: string | undefined;
   transactionHashStake?: string | undefined;
+  transactionHashUnstake?: string | undefined;
 }
 
 const ApproveSuccess: FC<ApproveSuccessProps> = ({
```

Comment:
This file works, and is fine for this PR, but I would suggest refactoring it to be a more generic component something more like like (obviously a lot more would be returned, but you get the idea):

```typescript

const TransactionTracker = ({
    txHash, 
    chainId = 1
  }: {
    txHash?: `0x{string}`, 
    chainId?: number, 
    successText: string
  })=> {
    const { isLoading, isSuccess } = useWaitForTransaction(txHash)
    const url = etherscanLink(txHash);

    return isLoading ? 
      <LoadingSpinner /> : 
      <a href={url}>{successText}</a>
}


```

Reply:
Can combine `TransactionFailed.tsx` into this by also pulling out `isError` from `useWaitForTransaction`

Reply:
I created a new issue for this to fix in a future PR. Issue: https://github.com/airswap/airswap-voter-rewards/issues/47